net/wireguard/src/opnsense/mvc/app/controllers/OPNsense/Wireguard/Api/ServiceController.php

```diff
@@ -62,8 +76,52 @@ public function showconfAction()
      */
     public function showhandshakeAction()
     {
+        $curtime = time();
         $backend = new Backend();
-        $response = $backend->configdRun("wireguard showhandshake");
-        return array("response" => $response);
+        $response_org = $backend->configdRun("wireguard showhandshake");
+        $resp_arr = array();
+
+        $pubnames = $this->getPubkeyNames();
+        $rp_lines = preg_split('/\r\n|\r|\n/', $response_org);
+        foreach($rp_lines as $line)
+        {
+            $line = trim($line);
+            if(empty($line)) continue;
+            $cols = preg_split('/[\s\t]{1,}/', $line);
+            if(count($cols) > 2)
+            {
+                $name = isset($pubnames[$cols[1]]) ? $pubnames[$cols[1]] : '<UNKNOWN>';
+                $timediff = $curtime - intval($cols[2]);
+                $date = !empty($cols[2]) ? ($timediff < 600 ? $timediff." sec. ago\t" : date('Y-m-d H:i:s', intval($cols[2]))) : "NEVER CONNECTED\t";
+                $extratab = empty($cols[2]) ? "\t\t" : "\t";
+                $resp_arr[] = $line. $extratab.$date."\t".$name;
+            } else {
+                $resp_arr[] = $line;
+            }
+        }
+        sort($resp_arr);
+
+        return array("response" => implode(PHP_EOL, $resp_arr));
+    }
+
+    /**
+     * build Dictionary pubkey => name
+     * @return array
+     */
+    private function getPubkeyNames()
+    {
+        $mdlclients = new Client();
+        $search = $mdlclients->getNodes();
+
+        $ret = array();
+        if(is_array($search['clients']['client']))
```

Comment:
I would check first if clients exists. It throws a warning / exception in case you access the client property of an undefined array